x64 SLE Micro instruction tweaks (#33)
suse-edge/suse-edge.github.io: docs/quickstart/slemicro-virt-install-x86_64.md

We have to create the image based and prepare the image with ignition and combustion files.
Basically we will use the following documents as reference to create the image changing the base SLEMicro image to be downloaded (**in this case will be SLE Micro x86_64**):

- Download the raw image file from the SUSE website at https://www.suse.com/download/sle-micro/
    - Select AMD64/Intel 64 architecture
    - Look for the raw file (I.e.- `SLE-Micro.x86_64-5.4.0-Default-GM.raw.xz`)
 
**NOTE:** You need to have a valid user on the SUSE site to be able to download the file.

> **If you are trying to download to a remote server, open this section for instructions:**

Reviewer: I don't think this is really needed :) Maybe you can add something like:

```suggestion
If you are trying to download to a remote server, you can use scp to copy that file to the server.
```

> <details>

Author: Oh my mistake, I actually forgot to ask about that

  <summary>Click here for instructions to get the raw file to remote server</summary>
  
**NOTES:** 
 - Be sure **NOT** to unzip the file before transfering it to your server  
 - Make sure to replace the paths as necessary
  
### Transfering file to server:
```
 scp -rp ~/PATH-TO-FILE/SLE-Micro.x86_64-5.4.0-Default-GM.raw.xz user@ip:~/PATH-TO-STORE-IN
```
</details>

- Access to <https://scc.suse.com/> to generate a registration code
- Butane, qemu-img and cdrtools installed (using zypper for example)
 ```bash

>   sudo zypper install butane qemu-tools xz

Reviewer: cdrtools package is missing I guess

Author: I found that I was able to go through the install (at least on tumbleweed) without it, but I also had to substitute qemu-img for the entire qemu-tools package, and it also needed the xz package

  passwd: 
    users: 
     - name: root 
       password_hash: "$y$j9T$/t4THH10B7esLiIVBROsE.$G1lyxfy/MoFVOrfXSnWAUq70Tf3mjfZBIe18koGOuXB" 
  EOF 
  ```

- Create a script combustion file as required. See the following
  example to register the SLE Micro instance to SUSE's SCC (use your
  own email/regcode) and to create a `/etc/issue.d/combustion` file

  ```
  cat << EOF > ${TMPDIR}/combustion/script  
  #!/bin/bash 
  # combustion: network 

  # Redirect output to the console 
  exec > >(exec tee -a /dev/tty0) 2>&1

  # Set hostname at combustion phase for SUSEConnect
  hostname slemicro

  # Registration 
  if ! which SUSEConnect > /dev/null 2>&1; then 
      zypper --non-interactive install suseconnect-ng 
  fi 

  SUSEConnect --email foobar@suse.com --url https://scc.suse.com --regcode YOURCODE 

  # Leave a marker 
  echo "Configured with combustion" > /etc/issue.d/combustion 
  EOF 
  ```

- Convert the butane config to ignition

  ```
  butane -p -o ${TMPDIR}/ignition/config.ign ${TMPDIR}/config.fcc
  ```

- Create an ISO file. It is requried for both ignition and combustion
  to work that the ISO is labeled as `ignition` (hence the -V
  parameter)

  ```

>   mkisofs -full-iso9660-filenames -o ignition-and-combustion.iso -V

Reviewer: Make this a single line just in case (see #34 )

  ignition ${TMPDIR}
  ```

- **Optional:** Remove the temporary folder

  ```
  rm -Rf ${TMPDIR}
  ```

## Create the VM
```bash
virt-install --name MyVM --memory 4096 --vcpus 4 --disk ./slemicro --import --cdrom ./ignition-and-combustion.iso --network default --osinfo detect=on,name=sle-unknown
```
**NOTES:** 

>  - Pass the -noautoconsole flag in case your console hangs on the installation, this will allow you to run other commands without CTRL-C intterupt

Reviewer: ```suggestion
 - Pass the -noautoconsole flag in case your console hangs on the installation, this will allow you to run other commands without CTRL-C interupt
```

>  - Pass the --debug flag if you run into issues

Reviewer: ```suggestion
 - Pass the `--debug` flag if you run into issues
```

 - If you run into an issue and you need to restart, or if you get an error saying that MyVM is already running, run this command:

```
 virsh destroy MyVM ; virsh undefine MyVM

> ```

Reviewer 2: I'd add it to a new sub-section named "delete VM" also. I missed that section in the first version, so maybe could be added now. The delete_vm.sh is basically doing that part. WDYT?
